src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -20,15 +20,48 @@
 
 namespace Stratis.Bitcoin.Features.Miner
 {
+    /// <summary>
+    /// <see cref="PosMinting"/> is used in order to generate new blocks. It involves a sort of lottery, similar to proof-of-work,
+    /// but the chances of winning this lottery is proportional to how many coins you are staking, not on hashing power.
+    /// </summary>
+    /// <remarks>
+    /// Staking is attempted only if the node is fully synchronized and connected to the network.
+    /// If not it will wait  till node is synced. Only transactions that were confirmed at least
+    /// <see cref="PosConsensusOptions.StakeMinConfirmations"/> blocks ago are eligible for staking.
+    /// </remarks>
+    /// <remarks>
+    /// The overall process for "attempting" to mine a PoS block looks like this:
+    /// 1) Create new block with transactions from <see cref="TxMempool"/>
+    /// 2) Get UTXOs (Unspent Transaction Output) that can participate in staking (have suitable depth).
+    /// 3) Split those UTXOs in subsets so each subset contains max of <see cref="utxoStakeDescriptionsPerCoinstakeWorker"/> transactions
+    /// and run CoinstakeWorker for each subset. Splitting is done only to achieve a good level of parallelism.
+    /// 4) CoinstakeWorker will try different versions of a block we are trying to mine (the only difference is the
+    /// timestamp which is tried in a time interval from now to now - searchInterval seconds.
```

Comment:
any formula should be inside `<c>` or `<code>`

I'm missing information about the mask here. This is quite important thing imho, even at big picture level. The mask is 0xf and it basically means that only unix timestamps that are divisible by 16 are valid candidates. So if you have block with timestamp X, you can't have a block with timestamp X+1 or X+2. but the next block must have X+16, X+32, X+48, ... see StakeTimestampMask